hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationServiceImpl.java

```diff
@@ -102,12 +102,15 @@ public WanReplicationServiceImpl(Node node) {
         };
     }
 
+    @Override
+    public boolean hasWanReplicationScheme(String wanReplicationScheme) {
+       return !(wanReplications.containsKey(wanReplicationScheme)
+                && node.getConfig().getWanReplicationConfig(wanReplicationScheme) == null);
+    }
```

Comment:
This logic is a bit confusing to me with the additional negative wrapping condition. It could be simplified according to De Morgan's laws as such.

```suggestion
    @Override
    public boolean hasWanReplicationScheme(String wanReplicationScheme) {
       return !wanReplications.containsKey(wanReplicationScheme) 
		|| node.getConfig().getWanReplicationConfig(wanReplicationScheme) != null;
    }
```

That being said, are we sure this logic is correct? return `true` when either `wanReplications` **does not** contain the key, or `getWanReplicationConfig` is `non-null`?

Reply:
There was a flaw in how I copied the original logic, thanks for catching. Updated